src/userspace/exec_monitor.cpp

```diff
@@ -81,4 +104,45 @@ int ExecMonitor::run()
 
 }
 
-int ExecMonitor::export_data() { return 0;} // TODO
\ No newline at end of file
+void ExecMonitor::export_data(process_info *p)
+{
+	if (export_format == STDOUT)
+	{
+		printf("%d\t%d\t%d\t%s\n", p->ppid, p->pid, p->tgid, p->name);
+		return;
+	}
+
+	file.open(filename, std::ios::out | std::ios::app | std::ios::binary);
```

Comment:
This also means that if the user provides an incorrect file, this will spam the stdout with the same error message over and over again. If you were to create file only once, you could also abort as soon as you realize that the file can not be opened.